GiEnJul/ClientApp/src/components/admin/overview/OverviewMacro.tsx

```diff
@@ -0,0 +1,191 @@
+import Recipient from "./Recipient";
+import { Button, Grid, Typography } from "@material-ui/core";
+import usePost from "../../../hooks/usePost";
+
+export interface SelectedConnectionType {
+  giverRowKey: string;
+  giverPartitionKey: string;
+  recipientRowKey: string;
+  recipientPartitionKey: string;
+}
+
+export interface GiverType {
+  email: string;
+  eventName: string;
+  fullName: string;
+  hasConfirmedMatch: Boolean;
+  isSuggestedMatch: Boolean;
+  location: string;
+  matchedRecipient?: string;
+  maxReceivers: Number;
+  partitionKey: string;
+  rowKey: string;
+  phoneNumber: string;
+  isSelected: Boolean;
+}
+
+export interface RecipientType {
+  contactEmail: string;
+  contactFullName: string;
+  contactPhoneNumber: string;
+  dessert: string;
+  dinner: string;
+  eventName: string;
+  familyMembers: PersonType[];
+  hasConfirmedMatch: Boolean;
+  institution: string;
+  isSuggestedMatch: Boolean;
+  location: string;
+  matchedGiver?: GiverType;
+  note: string;
+  partitionKey: string;
+  referenceId: string;
+  rowKey: string;
+  isSelected: Boolean;
+}
+export interface PersonType {
+  partitionKey: string;
+  rowKey: string;
+  wish: string;
+  age: Number;
+  gender: Number;
+}
+
+function OverviewMacro() {
+  const [selectedConnection, setSelectedConnection] =
+    useState<SelectedConnectionType>({
+      giverRowKey: "",
+      giverPartitionKey: "",
+      recipientRowKey: "",
+      recipientPartitionKey: "",
+    });
+  const [giverData, setGiverData] = useState<GiverType[] | []>([]);
+  const [recipientData, setRecipientData] = useState<RecipientType[] | []>([]);
+
+  async function fetchGivers() {
+    await fetch("api/admin/givers", {
+      method: "GET", // *GET, POST, PUT, DELETE, etc.
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then((response) => response.json())
+      .then((json) => setGiverData(json))
+      .catch((errorStack) => {
+        console.log(errorStack);
+      });
+  }
+  async function fetchRecipients() {
+    await fetch("./api/admin/recipients", {
+      method: "GET", // *GET, POST, PUT, DELETE, etc.
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then((response) => response.json())
+      .then((json) => setRecipientData(json))
+      .catch((errorStack) => {
+        console.log(errorStack);
+      });
+  }
+
+  useEffect(() => {
+    fetchRecipients();
+    fetchGivers();
+  }, []);
+
+  const handleGiverChange = (
+    newGiverRowKey: string,
+    newGiverPartitionKey: string
+  ) => {
+    setSelectedConnection((prev) => ({ ...prev, giverRowKey: newGiverRowKey }));
+    setSelectedConnection((prev) => ({
+      ...prev,
+      giverPartitionKey: newGiverPartitionKey,
+    }));
+    setGiverData(
+      giverData.map((item) =>
+        item.rowKey === newGiverRowKey && item.isSuggestedMatch === false
+          ? { ...item, isSelected: true }
+          : { ...item, isSelected: false }
+      )
```

Comment:
This can be shortended:
```suggestion
      giverData.map((item) =>
        ({ ...item, isSelected: (item.rowKey === newGiverRowKey && item.isSuggestedMatch === false)})
      )
```

Reply:
This applies below as well on line 129 `setRecipientData()`